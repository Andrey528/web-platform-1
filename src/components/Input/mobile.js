import Input from './view/mobile/Input';
export default Input;