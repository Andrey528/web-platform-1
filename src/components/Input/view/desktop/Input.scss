.input {
  padding-left: 1rem;
  box-sizing: border-box;
  border: none;
  border-bottom: 2px solid #5892ff;
  background: transparent;
  display: block;
  font-size: 1rem;
  width: 100%;
  min-height: 2rem;
  outline: none;
  color: #5892ff;

  &_sizable {
    &_default {
      height: 100%;
    }
    &_low {
      height: 2rem;
    }
    &_high {
      height: 3rem;
    }
  }

  &_type {
    &_date {
      font-size: 1rem;
      font-family: 'Segoe UI', sans-serif;
      &::-webkit-outer-spin-button,
      &::-webkit-inner-spin-button {
      -webkit-appearance: none;
      margin: 0;
      color: white;
      }
    }
    &_one-click {
      border-radius: 0;
      width: 4rem;
      text-align: center;
      padding-left: 0;
      height: 2.1429rem;;
      color: white
    }
  }
}