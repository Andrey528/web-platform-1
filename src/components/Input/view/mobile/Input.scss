.input {
  box-sizing: border-box;
  border: none;
  border-bottom: 2px solid #5892ff;
  background: transparent;
  display: block;
  font-size: 1rem;
  width: 100%;
  min-height: 2rem;
  outline: none;
  color: #fff;

  &::placeholder {
    color: #6C7883;
  }

  &_sizable {
    &_default {
      height: 100%;
    }
    &_low {
      height: 2rem;
    }
    &_high {
      height: 3rem;
    }
    &_rounded {
      border-radius: 1.17rem;
    }
  }

  &_type {
    &_date {
      font-size: 0.75rem;
      font-family: 'Segoe UI', sans-serif;
      &::-webkit-outer-spin-button,
      &::-webkit-inner-spin-button {
      -webkit-appearance: none;
      margin: 0;
      }
    }
    &_input-basket-modal {
      border-radius: 0;
      width: 5rem;
      font-size: 1rem;
    }
  }
}