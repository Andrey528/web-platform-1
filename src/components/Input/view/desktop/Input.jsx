import React from 'react';
import block from 'bem-cn';
import PropTypes from 'prop-types';

import './Input.scss';

const Input = ({ value, name, callBack, placeholder = '', size = 'default', type = 'text', isRequired, ...props }) => {
  const b = block('input');
  return (
    <input
      type={type}
      name={name}
      value={value}
      placeholder={placeholder || ''}
      onChange={callBack}
      className={b({ sizable: size, type })}
      required={isRequired}
      {...props}
      />
  );
};

Input.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  name: PropTypes.string.isRequired,
  callBack: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  size: PropTypes.string,
  type: PropTypes.string,
  isRequired: PropTypes.bool,
};

export default Input;