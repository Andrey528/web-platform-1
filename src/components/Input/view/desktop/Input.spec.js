import React from 'react';
import { shallow } from 'enzyme';
import block from 'bem-cn';

import Input from './Input';

describe('Input tests', () => {
  const b = block('input');
  const props = {
    value: 'Jopa',
    callBack: jest.fn(),
    name: 'Jopa',
    placeholder: 'Test placeholder',
    type: 'Jopa',
    isRequired: true,
    size: 'low',
  };

  describe('Input with value', () => {
    const input = shallow(<Input {...props} />);

    it('render properly', () => {
      expect(input).toMatchSnapshot();
    });

    it('render value', () => {
      expect(input.find('input').props().value).toEqual(props.value);
    });

    it('render name', () => {
      expect(input.find('input').props().name).toEqual(props.name);
    });

    it('render placeholder', () => {
      expect(input.find('input').props().placeholder).toEqual(props.placeholder);
    });

    it('render type', () => {
      expect(input.find('input').props().type).toEqual(props.type);
    });

    it('render required', () => {
      expect(input.find('input').props().required).toEqual(props.isRequired);
    });
  });

  describe('Input have modificators', () => {
    const defaultProps = {
      size: undefined,
      type: undefined,
    };
    it('Have default modificators', () => {
      const className = b({ sizable: 'default' }, { type: 'text' });
      const input = shallow(<Input {...{ ...props, ...defaultProps }} />);

      expect(input.find('input').props().className.toString()).toEqual(className.toString());
    });

    it('Have different modificators', () => {
      const className = b({ sizable: props.size }, { type: props.type });
      const input = shallow(<Input {...props} />);

      expect(input.find('input').props().className.toString()).toEqual(className.toString());
    });
  });

  describe('Callback test', () => {
    it('onChange test', () => {
      const input = shallow(<Input {...props} />);

      input.find('input').simulate('change', 'test');
      expect(props.callBack).toBeCalledWith('test');
    });
  });
});