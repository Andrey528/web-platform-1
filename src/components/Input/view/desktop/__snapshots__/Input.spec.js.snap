// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests Input with value render properly 1`] = `
<input
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  name="Jopa"
  onChange={[MockFunction]}
  placeholder="Test placeholder"
  required={true}
  type="Jopa"
  value="Jopa"
/>
`;
